### PR TITLE
PS-7702 Document jemalloc memory allocation profiling on PS 8.0 (8.0)

### PR DESCRIPTION
**Pull request by patrickbirch**
Changes to be committed:
	modified:   source/index.rst
	new file:   source/performance/jemalloc-profiling.rst

### CODE REVIEW

File: doc/source/performance/jemalloc-profiling.rst [hunk 1]
```diff
@@ -0,0 +1,154 @@
+.. _jemalloc-profiling:
+
+=================================================
+Jemalloc Memory Allocation Profiling
+=================================================
+
+Implemented in :rn:`8.0.25-15`, you are able to profile the jemalloc memory allocation.
```
Comment (george-lorch) [inline doc/source/performance/jemalloc-profiling.rst]: I think we need to be clearer and indicate that this is only present and useful if the server is running with the jemalloc library.  @ldonoso ?

Comment (patrickbirch) [reply doc/source/performance/jemalloc-profiling.rst]: Fixed. Completely rewritten.

**Review by george-lorch — APPROVED** [review]

File: doc/source/performance/jemalloc-profiling.rst (hunk 1 above)
Comment (ldonoso) [reply doc/source/performance/jemalloc-profiling.rst]: Sorry for the delay @georgelorchpercona, yes, I agree, thanks for the change @patrickbirch

File: doc/source/performance/jemalloc-profiling.rst [hunk 2]
```diff
@@ -0,0 +1,218 @@
+.. _jemalloc-profiling:
+
+=================================================
+Jemalloc Memory Allocation Profiling
+=================================================
+
+Implemented in :rn:`8.0.25-15`, |Percona Server| can take advantage of the  memory-profiling ability of the jemalloc allocator. This ability provides a method to investigate memory-related issues.
```
Comment (ldonoso) [inline doc/source/performance/jemalloc-profiling.rst]: Please, mind the two spaces on
```suggestion
Implemented in :rn:`8.0.25-15`, |Percona Server| can take advantage of the memory-profiling ability of the jemalloc allocator. This ability provides a method to investigate memory-related issues.
```

File: doc/source/performance/jemalloc-profiling.rst [hunk 3]
```diff
@@ -0,0 +1,218 @@
...
+
+
+Requirements
+----------------
+
+This memory-profiling requires :ref:`jemalloc_detected`. This read-only variable returns ``true`` if jemalloc with the profiling-enabled option is detected.
```
Comment (ldonoso) [inline doc/source/performance/jemalloc-profiling.rst]:
```suggestion
This memory-profiling requires :ref:`jemalloc_detected`. This read-only variable returns ``true`` if jemalloc with the profiling-enabled option is being used by Percona Server. 
```

File: doc/source/performance/jemalloc-profiling.rst [hunk 4]
```diff
@@ -0,0 +1,218 @@
...
+=================================================
+
+Implemented in :rn:`8.0.25-15`, |Percona Server| can take advantage of the  memory-profiling ability of the jemalloc allocator. This ability provides a method to investigate memory-related issues.
+
+
+Requirements
+----------------
+
+This memory-profiling requires :ref:`jemalloc_detected`. This read-only variable returns ``true`` if jemalloc with the profiling-enabled option is detected. 
+
```
Comment (ldonoso) [inline doc/source/performance/jemalloc-profiling.rst:14]: Please, I think it'd be a good idea to add a paragraph to introduce the topic. i.e. "Steps to make Percona Server use jemalloc with profiling enabled"

File: doc/source/performance/jemalloc-profiling.rst [hunk 5]
```diff
@@ -0,0 +1,230 @@
...
+    SELECT * FROM malloc_stats_totals;
+    +----+------------+------------+------------+-------------+------------+
+    | id | ALLOCATION | MAPPED     | RESIDENT   | RETAINED    | METADATA   |
+    +----+------------+------------+------------+-------------+------------+
+    |  1 | 390977528  | 405291008  | 520167424  | 436813824   | 9933744    |
+    +----+------------+------------+------------+-------------+------------+
+    1 row in set (0.00 sec)
+
+The :ref:`malloc_stats` table returns the cumulative totals, in bytes, of allocation requests and allocation returns. The command takes no parameters and returns the results as a table.
```
Comment (ldonoso) [inline doc/source/performance/jemalloc-profiling.rst]:
```suggestion
The :ref:`malloc_stats` table returns the cumulative totals, in bytes, of several statistics per type of arena. The command takes no parameters and returns the results as a table.
```

File: doc/source/performance/jemalloc-profiling.rst [hunk 6]
```diff
@@ -0,0 +1,230 @@
...
+    +--------+-------------+-------------+-------------+-------------+
+    | TYPE   | ALLOCATED   | NMALLOC     | NDALLOC     | NRESQUESTS  |
+    +--------+-------------+-------------+-------------+-------------+
+    | small  | 23578872    | 586156      | 0           | 2649417     |
+    | large  | 367382528   | 2218        | 0           | 6355        |
+    | huge   | 0           | 0           | 0           | |
+    +--------+-------------+-------------+-------------+-------------+
+    3 rows in set (0.00 sec)
+
+The profiling samples the ``malloc()`` calls and stores the sampled stack traces in a separate location in memory. These samples can be dumped into the filesystem. A dump returns a detailed view of the state of the memory.
```
Comment (ldonoso) [inline doc/source/performance/jemalloc-profiling.rst:95]: Maybe I would create a sub section for this i.e. `Dumping the memory profile`

**Review by ldonoso — CHANGES_REQUESTED** [review]

File: doc/source/performance/jemalloc-profiling.rst (hunk 2 above)
Comment (patrickbirch) [reply doc/source/performance/jemalloc-profiling.rst]: Fixed.

File: doc/source/performance/jemalloc-profiling.rst (hunk 3 above)
Comment (patrickbirch) [reply doc/source/performance/jemalloc-profiling.rst]: updated.

File: doc/source/performance/jemalloc-profiling.rst (hunk 5 above)
Comment (patrickbirch) [reply doc/source/performance/jemalloc-profiling.rst]: updated.

File: doc/source/performance/jemalloc-profiling.rst (hunk 6 above)
Comment (patrickbirch) [reply doc/source/performance/jemalloc-profiling.rst:95]: Added

File: doc/source/performance/jemalloc-profiling.rst (hunk 4 above)
Comment (patrickbirch) [reply doc/source/performance/jemalloc-profiling.rst:14]: Done.

**Review by ldonoso — APPROVED** [review]

**Review by george-lorch — APPROVED** [review]

**Review by george-lorch — APPROVED** [review]

File: doc/source/performance/jemalloc-profiling.rst [hunk 7]
```diff
@@ -0,0 +1,234 @@
...
+    Ensure the ``libjemalloc.so`` exists in the LD path.
+
+Use |Percona Server| with jemalloc with profiling enabled
+----------------------------------------------------------
+
+To detect if jemalloc is set, run the following command:
+
+.. sourcecode::  bash
+
+    jemalloc_detected ON
```
Comment (ghost) [inline doc/source/performance/jemalloc-profiling.rst]: I think this should be `SELECT @@jemalloc_detected;`?

Comment (patrickbirch) [reply doc/source/performance/jemalloc-profiling.rst]: Fixed.

**Review by ldonoso — APPROVED** [review]

**Pull request merged**